src/content/chainlink-functions/resources/simulation.mdx

```diff
@@ -0,0 +1,78 @@
+
+Currently, there are several options for simulating a request:
+
+- [Chainlink Functions playground](https://functions.chain.link/playground).
+
+- [Chainlink Functions Hardhat Starter Kit](https://github.com/smartcontractkit/functions-hardhat-starter-kit): Use the `npx hardhat functions-simulate-script` command to simulate your Functions JavaScript source code.
+
+- [Chainlink Functions Toolkit NPM package](https://github.com/smartcontractkit/functions-toolkit): Import this NPM package into your JavaScript/TypeScript project, then use the `simulateScript` function to simulate your Functions JavaScript source code.
+
+## Chainlink Functions playground
+
+<Aside type="caution">
+  The simulation is run on the browser. There may be differences to the real Chainlink Functions environment. For
+  instance, you are limited to vanilla JavaScript and cannot use all the functionalities of Chainlink Functions, such as
+  Imports of external modules. To access all the functionalities, use the [Chainlink Functions Hardhat Starter
+  Kit](https://github.com/smartcontractkit/functions-hardhat-starter-kit) or [Chainlink Functions Toolkit NPM
+  package](https://github.com/smartcontractkit/functions-toolkit).
+</Aside>
+
+To explore the capabilities of Chainlink Functions, visit the [Chainlink Functions Playground](https://functions.chain.link/playground). This playground can simulate Chainlink Functions within the browser, call APIs, and execute demo requests.
+To use the playground, enter any source code, arguments, and secrets you would like to use. Click the **Run code** button to view the output.
+
+<ClickToZoom src="/images/chainlink-functions/functions-playground.png" />
+
+## Chainlink Functions Hardhat Starter Kit
+
+This repository comes preconfigured with [Hardhat](https://hardhat.org/) and the [Chainlink Functions Toolkit NPM package](https://github.com/smartcontractkit/functions-toolkit), allowing you to quickly get started with Functions.
+
+To simulate:
+
+1.  In a terminal, clone the [functions-hardhat-starter-kit repository](https://github.com/smartcontractkit/functions-hardhat-starter-kit) and change to the `functions-hardhat-starter-kit` directory.
+
+    ```shell
+    git clone https://github.com/smartcontractkit/functions-hardhat-starter-kit && \
+    cd functions-hardhat-starter-kit
+    ```
+
+1.  Run `npm install` to install the dependencies.
+
+    ```shell
+    npm install
+    ```
+
+1.  For simulation, you don't need to set up the environment variables. Go ahead and run `npx hardhat functions-simulate-script` to simulate the [calculation-example.js](https://github.com/smartcontractkit/functions-hardhat-starter-kit/blob/main/calculation-example.js) JavaScript source code.
+
+    ```shell
+    npx hardhat functions-simulate-script
+    ```
+
+    Result:
+
+    ```text
+    secp256k1 unavailable, reverting to browser version
+
+
+    Response returned by script during local simulation: 1003757
```

Comment:
Let's add some context about what this response means - maybe introduce the fact that it's a calculator example at some point before the Aside text below